util/ranger/checker.go

```diff
@@ -98,6 +98,10 @@ func (c *conditionChecker) checkScalarFunction(scalar *expression.ScalarFunction
 }
 
 func (c *conditionChecker) checkLikeFunc(scalar *expression.ScalarFunction) bool {
+	_, collation, _ := scalar.CharsetAndCollation(scalar.GetCtx())
```

Comment:
LGTM for this check(according to your explanations in the PR description).

But I don't get the point why we got a full index scan for `LIKE`, if the collations are not compatible?